app/services/analytics-value-service.js

```diff
@@ -1,18 +1,49 @@
+  metrics:[ "tSegmentDuration", "tConversationDuration", "oTotalCriticalScore", "oTotalScore", "nEvaluations", "tAbandon", "tIvr", "tAnswered", "tAcd", "tTalk", "tHeld", "tTalkComplete", "tHeldComplete", "tAcw", "tHandle", "tWait", "tAgentRoutingStatus", "tOrganizationPresence", "tSystemPresence", "tUserResponseTime", "tAgentResponseTime", "nOffered", "nOverSla", "nTransferred", "nOutboundAttempted", "nOutboundConnected", "nOutboundAbandoned", "nError", "oServiceTarget", "oServiceLevel", "tActive", "tInactive", "oActiveUsers", "oMemberUsers", "oActiveQueues", "oMemberQueues", "oInteracting", "oWaiting", "oOnQueueUsers", "oOffQueueUsers", "oUserPresences", "oUserRoutingStatuses" ].sort(),
+  groupBy:[ "conversationId", "sessionId", "mediaType", "queueId", "userId", "participantId", "participantName", "direction", "wrapUpCode", "wrapUpNote", "interactionType", "requestedRoutingSkillId", "requestedLanguageId", "purpose", "participantType", "segmentType", "disconnectType", "errorCode", "conversationEnd", "segmentEnd", "externalContactId", "externalOrganizationId", "stationId", "edgeId", "dnis", "ani", "outboundCampaignId", "outboundContactId", "outboundContactListId", "monitoredParticipantId", "sourceSessionId", "destinationSessionId", "sourceConversationId", "destinationConversationId", "remoteNameDisplayable", "sipResponseCode", "q850ResponseCode", "conference", "groupId", "roomId", "addressFrom", "addressTo", "subject", "peerId", "scriptId", "evaluationId", "evaluatorId", "contextId", "formId", "formName", "eventTime", "systemPresence", "organizationPresenceId", "routingStatus" ].sort(),
+  
+  propertyTypes:["", "bool", "integer", "real", "date", "string", "uuid"].sort(),
+  operators:["matches", "exists", "notExists"],
+  mediaTypes:["", "voice", "chat", "email", "callback", "screenshare", "cobrowse"].sort(),
+  aggregationTypes:['termFrequency', 'numericRange'],
+  filteredMetrics: function(filter){
+    var patt = new RegExp(filter);
 
-        return this.get('metrics').filter(function(item){
-            return patt.test(item);
-          });
-    }
+    return this.get('metrics').filter(function(item){
+      return patt.test(item);
+    });
+  },
+  init() {
+    let that = this;
+    that.set('metrics',['metric A', 'metric B']);
+    var swaggerUrl = '/swagger-schema/publicapi-v2-latest.json';
+    $.getJSON({
```

Comment:
this might be better in its own service that way if we need the swagger somewhere else the service will have it cached.  Then setup an observer so when the service gets the swagger it computes everything in the done func

Reply:
Ok, will refactor for that.